crates/polars-core/src/chunked_array/logical/datetime.rs

```diff
@@ -70,14 +80,58 @@ impl LogicalType for DatetimeChunked {
             },
             #[cfg(feature = "dtype-date")]
             Date => {
+                let has_timezone = self.time_zone().is_some();
+                let timestamp_to_datetime = match self.time_unit() {
+                    TimeUnit::Milliseconds => timestamp_ms_to_datetime,
+                    TimeUnit::Microseconds => timestamp_us_to_datetime,
+                    TimeUnit::Nanoseconds => timestamp_ns_to_datetime,
+                };
+                let datetime_to_timestamp = match self.time_unit() {
+                    TimeUnit::Milliseconds => datetime_to_timestamp_ms,
+                    TimeUnit::Microseconds => datetime_to_timestamp_us,
+                    TimeUnit::Nanoseconds => datetime_to_timestamp_ns,
+                };
                 let cast_to_date = |tu_in_day: i64| {
-                    let mut dt = self
-                        .phys
-                        .apply_values(|v| v.div_euclid(tu_in_day))
+                    let values = {
+                        #[cfg(feature = "timezones")]
+                        {
+                            if has_timezone {
```

Comment:
a pattern that's used elsewhere is something like
```
match self.dtype() {
    #[cfg(feature = "timezones")]
    Datetime(_, Some(tz)) => ...
    _ => ...
```
does something like that work here, to avoid duplicating `self.phys.apply_values(|v| v.div_euclid(tu_in_day))`?